howdoi/howdoi.py

```diff
@@ -410,29 +409,41 @@ def _get_answers(args):
     if not question_links:
         return False
 
-    answers = []
-    initial_position = args['pos']
-    multiple_answers = (args['num_answers'] > 1 or args['all'])
+    initial_position = args['pos'] - 1
+    final_position = initial_position + args['num_answers']
+    question_links = question_links[initial_position: final_position]
+
+    with Pool() as pool:
+        answers = pool.map(partial(_get_answer_worker, args=args), question_links)
 
-    for answer_number in range(args['num_answers']):
-        current_position = answer_number + initial_position
-        args['pos'] = current_position
-        link = get_link_at_pos(question_links, current_position)
-        answer = _get_answer(args, question_links)
-        if not answer:
-            continue
-        if not args['link'] and not args['json_output'] and multiple_answers:
-            answer = ANSWER_HEADER.format(link, answer, STAR_HEADER)
-        answer += '\n'
-        answers.append({
-            'answer': answer,
-            'link': link,
-            'position': current_position
-        })
+    for idx, _ in enumerate(answers):
+        answers[idx]['position'] = idx + 1
```

Comment:
This is clever, I was wondering how you were doing to keep them in order.

Reply:
Yeah the threadpool maps elements one-one so ordering was preserved, but indexing from inside a parallel section is a recipe for disaster - so I had to resort to assigning position labels after the fact :smile: